fedimint-core/src/invite_code.rs

```diff
@@ -261,4 +270,64 @@ mod tests {
             ]
         );
     }
+
+    #[test]
+    fn invite_code_v2_encode_base64_roundtrip() {
+        let invite_code = InviteCodeV2 {
+            id: FederationId(sha256::Hash::from_slice(&[0; 32]).unwrap()),
+            peers: BTreeMap::from_iter([(
+                PeerId::from(0),
+                SafeUrl::parse("https://mint.com").expect("Url is valid"),
+            )]),
+            api_secret: None,
+        };
+
+        let encoded = invite_code.encode_base64();
+        let decoded = InviteCodeV2::decode_base64(&encoded).expect("Failed to decode");
+
+        assert_eq!(invite_code, decoded);
+
+        InviteCode::from_str(&encoded).expect("Failed to decode to legacy");
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Encodable, Decodable)]
+pub struct InviteCodeV2 {
+    pub id: FederationId,
+    pub peers: BTreeMap<PeerId, SafeUrl>,
+    pub api_secret: Option<String>,
```

Comment:
Can you annotate this to use the default value (`None`) if not present and not serialize if not present? https://stackoverflow.com/a/53900684